### DISCUSSION TITLE
Epoch stuck

### DISCUSSION THREAD

**Discussion by helmihasibuan** [teamalgoritma/community#620] [Q&A] [answered]
epoch berhenti di 5, lalu loading gak berhenti2. kira kira kenapa ya kak? 
terima kasih

![image](https://user-images.githubusercontent.com/65268735/231055332-88492bf5-9a17-4e01-94a9-3aa5749f2fd5.png)

**Comment by amaliasf** [comment, marked as answer]
Dear Pak @helmihasibuan ,

Terkait kendala yang Bapak hadapi bisa di cek di dikusi ini Pak https://github.com/teamalgoritma/community/discussions/319 . Sepertinya memiliki permasalahan yang serupa.

Semoga informasi ini dapat membantu. Bila masih terkendala silahkan diinfokan ya Pak.

Kalau ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Reply by DeanAlexander27** [reply]
halo Pak. saya mengalami hal yang sama stuck ditengah dengan jumlah epoch 10 dengan data set yang di scale 128x128
tetapi dari hasil diskusi yang diarahkan sepertinya tidak ada jalan keluar..

saya tidak paham yang dijadikan jawaban ini itu apa ..

Instead of using fit( ), use fit_generator( ). In fit_generator(), you don't pass the x and y directly, instead, they come from a generator. As it is written in [keras documentation](https://keras.io/models/sequential/), the generator is used when you want to avoid duplicate data when using multiprocessing. This is for practical purposes when you have a large dataset.

**Reply by nadyadtm** [reply]
Dear Pak @DeanAlexander27 dan Pak @helmihasibuan 

Berdasarkan jawaban pada diskusi yang sebelumnya, silakan pada saat pelatihan dicoba mengganti `fit()` menjadi `fit_generator()`. Untuk isinya tidak perlu diganti. Apakah hal tersebut sudah dicoba?

**Reply by DeanAlexander27** [reply]
baik ka @nadyadtm .. untuk penggunaan fit() dan fit_generator() itu apakah ada perbedaan nya kah ?
setelah saya telusuri dengat keras cheat sheet dijelaskan it_generator() *Fits the model on data yielded batch by-batch by a generator*
bagaimana cara kerjanya ya ?
apakah berbeda jauh dengan cara kerja fit() ?

terima kasih

**Reply by nadyadtm** [reply]
Baik Pak @DeanAlexander27, 

Singkatnya seperti ini. fit() biasa digunakan ketika kita tidak akan melakukan augmentasi data dengan image data generator. Namun ketika akan menggunakan image data generator, kita dapat menggunakan fit_generator(). 

Berikut sebuah link yang menjelaskan tentang perbedaan fit() dan fit_generator() lebih lanjut. Semoga dapat membantu
- https://www.geeksforgeeks.org/keras-fit-and-keras-fit_generator/

**Reply by DeanAlexander27** [reply]
Dear ka @nadyadtm 
ohh baik.. sangat jelas.. terima kasih ...
